Add sidebar for club admin pages

SillyRobot883 opened the pull request:
I added a sidebar that replaces both the header and footer. So you should only use the sidebar in the following pages:

- /club-dashboard @SaifAlotaibie 
- /event-creation @AziizDev 

> I'll make the necessary changes to both the event-creation and club-dashboard branches 


Here's an example of sidebar integrated into the event-creation page:
![image](https://github.com/user-attachments/assets/10086c2d-f92a-4802-bba6-a993aa663b8e)


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

## Release Notes

- **New Features**
    - Introduced a new sidebar navigation system.
    - Added a redesigned landing page with club statistics.
    - Implemented a new club dashboard route.
    - Enhanced user authentication flow.
    - Added a dedicated login interface for users.

- **Improvements**
    - Migrated to EJS layouts for more consistent page rendering.
    - Updated UI components with Bootstrap styling.
    - Improved responsive design for mobile and desktop.
    - Enhanced error handling and user feedback mechanisms.
    - Improved styling and layout for various user interface components.

- **Design Updates**
    - Implemented a modern, clean design with Bootstrap components.
    - Added custom CSS for improved visual consistency.
    - Introduced new typography and color schemes.
    - Created more intuitive navigation and user interfaces.

- **Performance**
    - Optimized routing and middleware.
    - Streamlined authentication checks.
    - Improved code organization and maintainability.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

SillyRobot883 committed:
feat(sidebar): add sidebar for club admin pages

SillyRobot883 committed:
rebase to dev

SillyRobot883 committed:
fix(sidebar): fix styling and layout

AFAskar reviewed (changes requested):
Use bootstrap classes instead of using custom css
That will make responsiveness easier and make the code easier to understand just from the markup

SaifAlotaibie commented:
i have done of this

SillyRobot883 committed:
feat(sidebar): convert styling to bootstrap

SillyRobot883 committed:
fix(sidebar): change الرئيسية to لوحة التحكم

SillyRobot883 committed:
feat(layout): add a base layout

SillyRobot883 committed:
feat(frontend): update app.js to follow layout

SillyRobot883 committed:
fix(frontend): update already existing pages to be part of layout

SillyRobot883 committed:
refactor(frontend): update the rest of pages and remove redundant code

SillyRobot883 committed:
feat(sidebar): add user management button

SillyRobot883 committed:
refactor(frontend): refactor of profile.ejs

SillyRobot883 committed:
fix(sidebar): disable user management option and change styling

SillyRobot883 committed:
fix(layout): fix undefined currentPage param

SillyRobot883 committed:
rebase sidebar branch

SillyRobot883 committed:
fix(routes): move profile routes from app.js to profile.js and profileController.js

SillyRobot883 committed:
fix(app.js): rename feed to events

SillyRobot883 committed:
fix(profile): fix profile controller tests

abo3skr2019 commented on backend/routes/profile.js:
```diff
@@ -5,19 +5,42 @@ const {validateProfileInput} = require('../middleware/ProfileValidation');
 
 // Render the profile update form
-router.get('/update-profile', isAuthenticated, profileController.renderUpdateProfileForm);
+router.get('/update-profile', isAuthenticated, (req, res) => {
```
The Rendering Code should be on the Controller

abo3skr2019 commented on backend/routes/profile.js:
```diff
@@ -5,19 +5,42 @@ const {validateProfileInput} = require('../middleware/ProfileValidation');
 
 // Render the logged-in user's profile
-router.get('/profile', isAuthenticated, profileController.renderProfile);
+router.get('/profile', isAuthenticated, (req, res) => {
```
Move it back to the Controller

abo3skr2019 commented on backend/routes/profile.js:
```diff
@@ -5,19 +5,42 @@ const {validateProfileInput} = require('../middleware/ProfileValidation');
+});
 
+// Render delete account page
 router.get('/delete-account', isAuthenticated, (req, res) => {
```
Create a function in the profile controller for rendering the form 
it should be named
**renderDeleteAccount**

abo3skr2019 commented on frontend/views/layouts/base-layout.ejs:
```diff
@@ -0,0 +1,48 @@
+    <link href="/css/partials/background.css" rel="stylesheet">
+    <%- typeof HeaderOrSidebar !== 'undefined' && HeaderOrSidebar === 'sidebar' ? '<link href="/css/partials/sidebar.css" rel="stylesheet">' : '<link href="/css/partials/header.css" rel="stylesheet">' %>
+    <%- typeof extraCSS !== 'undefined' ? extraCSS : '' %>
+    <style>
```
Use Bootstrap Classes instead of a Style tag since you already imported bootstrap to the layout

abo3skr2019 commented on frontend/views/partials/sidebar.ejs:
```diff
@@ -0,0 +1,38 @@
+    </div>
+</div>
+<style>
+    .sidebar {
```
remove this and instead use a bootstrap class

abo3skr2019 commented on frontend/views/partials/sidebar.ejs:
```diff
@@ -0,0 +1,38 @@
+<div class="sidebar bg-white vh-100 position-fixed top-0 end-0 border-start" role="navigation" aria-label="القائمة الجانبية">
```
```suggestion
<div class="sidebar bg-white vh-100 position-fixed top-0 end-0 border-start overflow-hidden" role="navigation" aria-label="القائمة الجانبية">
```

AFAskar reviewed (changes requested):
Code Semantics issues only.
functionality not tested Yet !

SillyRobot883 committed:
refactor(profile): move rendering from routes to controller | add new renderDeleteAccount func

SillyRobot883 committed:
refactor(frontend): switch style tags to bootstrap classes

SillyRobot883 commented:
> Code Semantics issues only. functionality not tested Yet !

I pushed a fix for the issues. Update the profileController.test.js please

SillyRobot883 committed:
fix(layout): fix position of footer

SillyRobot883 committed:
fix(sidebar): correct sidebar navigation highlighting for club dashboard page

abo3skr2019 commented on backend/controllers/profileController.js:
```diff
@@ -44,12 +45,19 @@ const renderProfile = async (req, res) => {
             return res.status(404).send('User not found');
         }
-        res.render('profile', { user });
+        res.render('profile', {
```
This is causing the test to fail so either rewrite it to pass the test or rewrite the test to work with this new code 
if you rewrite the test you must still be keeping the essence of the test don't just do a automatice pass

SillyRobot883 committed:
feat(header): dynamically change header based on login status | resolves #43

SillyRobot883 committed:
fix(profileTest): update the profile tests

SillyRobot883 commented:
pushed an update for the profile tests, and I resolved #43

SillyRobot883 committed:
fix(app.js): move user middleware after database connection to ensure req.user is populated

SillyRobot883 committed:
refactor(profile): move profile render to Controller

SillyRobot883 committed:
feat(header): add relevant buttons based on user role

SillyRobot883 committed:
feat(header): add list of student clubs tab

SillyRobot883 committed:
feat(clubs): a general overhaul of all the /clubs pages

SillyRobot883 committed:
refactor(tests): add updates to profile Controller tests

SillyRobot883 commented:
### Changes Overview
1. **Overhaul of the design for the pages under clubs**
   - updated the pages in the clubs directory to match the layout and the design of the website
 
2. #43 **done**

3. **Fixed the `profileController.test.js` Test**:
   - Updated the test to match the actual implementation of `renderUpdateProfileForm`.
   - The test now checks for additional properties (`title`, `HeaderOrSidebar`, `extraCSS`, `currentPage`) passed to `res.render`.

SillyRobot883 committed:
refactor(frontend): update error page design

abo3skr2019 commented on frontend/views/partials/sidebar.ejs:
add the edit club and club details buttons to it

SillyRobot883 committed:
fix(clubs): make the assign-admin modal under club-details.ejs

SillyRobot883 committed:
fix(assign-admin): enhance function security and validiation

SillyRobot883 committed:
refactor(club.test): update tests to match club controller changes

SillyRobot883 committed:
fix(login.ejs): update icon classes

SillyRobot883 committed:
fix(login.ejs): disable unimplemented GitHub Auth button

SillyRobot883 committed:
refactor(clubController): add renderDashboard and renderEventCreation

SillyRobot883 committed:
refactor(clubRoutes): add new routes for club dashboard

SillyRobot883 committed:
refactor(frontend): update header and sidebar club dashboard buttons

SillyRobot883 commented:
### Changes


- moved `/club-dashboard` route to `/clubs/club.UUID/dashboard` 
- moved `/event-creation` route to `/clubs/club.UUID/dashboard/event-creation`
- added 2 new functions to `clubController.js`: renderDashboard, renderEventCreation
- refactored `clubRoutes.js` to easily distinguish between public and protected routes, and added the new functions from `clubController.js` 
- updated the club dashboard button on both the sidebar and the header so it redirects to `/clubs/club.UUID/dashboard` instead of `/club-dashboard`

SillyRobot883 committed:
refactor(backend): remove the eventCreation

SillyRobot883 committed:
refactor(app.js): move club admin middleware to backend

SillyRobot883 committed:
refactor(frontend): remove uncessary comments

SillyRobot883 committed:
fix(app.js): remove reduandant club admin middleware

SillyRobot883 committed:
fix(app.js): remove uncessary comments

SillyRobot883 committed:
refactor(frontend): use bootstrap wherever possible

SillyRobot883 committed:
feat(frontend): add home page inital design

SillyRobot883 committed:
refactor(update-club): move update-club functionality to /dashboard

SillyRobot883 committed:
fix(footer): revert back changes

SillyRobot883 committed:
feat(club-details): add a dashboard button for Admins

AFAskar reviewed (approved):
lgtm
